numpy/tests/typing/reveal/dtype.py

```diff
@@ -0,0 +1,9 @@
+import numpy as np
+
+reveal_type(np.dtype(np.float64))  # E: numpy.dtype[numpy.float64*]
+reveal_type(np.dtype(np.int64))  # E: numpy.dtype[numpy.int64*]
+
+# Uh oh, this shouldn't be ok
```

Comment:
Some examples that would need to be sorted out.

Reply:
Looking good thus far!
In all likelihood we'll have to go big on the `@overload`s (~1 per initiable `generic` subclass) to remedy the issues below.